src/client/QXmppOutgoingClient.cpp

```diff
@@ -142,7 +143,7 @@ void QXmppOutgoingClientPrivate::connectToHost(const QString &host, quint16 port
 
     // override CA certificates if requested
     if (!config.caCertificates().isEmpty())
-        q->socket()->setCaCertificates(config.caCertificates());
+        QSslConfiguration::defaultConfiguration().setCaCertificates(config.caCertificates());
```

Comment:
What does this change?

Reply:
>  Port deprecated method setCaCertificates of QSslSocket 

If `setCaCertificates` is deprecated, why do you still use it?

Reply:
Nevermind

Reply:
Moreover, doesn't this affect all other sockets in the application using qxmpp? That might be undesirable.

Reply:
> If `setCaCertificates` is deprecated, why do you still use it?

This is setCaCertificates of QSslConfiguration not QSslSocket

Reply:
But you shouldn't change the *global* default:
```C++
QSslConfiguration sslConfig;
sslConfig.setCaCertificates(config.caCertificates());
q->socket()->setSslConfiguration(sslConfig);
```

Reply:
> But you shouldn't change the _global_ default:
> 
> ```c++
> QSslConfiguration sslConfig;
> sslConfig.setCaCertificates(config.caCertificates());
> q->socket()->setSslConfiguration(sslConfig);
> ```

I noticed that and already pushed a fix